rolling-shutter/keyperimpl/gnosis/syncmonitor.go

```diff
@@ -0,0 +1,61 @@
+package gnosis
+
+import (
+	"context"
+	"time"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+
+	"github.com/shutter-network/rolling-shutter/rolling-shutter/keyperimpl/gnosis/database"
+	"github.com/shutter-network/rolling-shutter/rolling-shutter/medley/service"
+)
+
+const (
+	checkInterval = 30 * time.Second
+)
+
+type SyncMonitor struct {
+	DBPool *pgxpool.Pool
+}
+
+func (s *SyncMonitor) Start(ctx context.Context, runner service.Runner) error {
+	runner.Go(func() error {
+		return s.runMonitor(ctx)
+	})
+
+	return nil
+}
+
+func (s *SyncMonitor) runMonitor(ctx context.Context) error {
+	var lastBlockNumber int64
+	db := database.New(s.DBPool)
+
+	for {
+		select {
+		case <-time.After(checkInterval):
+			record, err := db.GetTransactionSubmittedEventsSyncedUntil(ctx)
+			if err != nil {
+				log.Warn().Err(err).Msg("error fetching block number")
+				continue
```

Comment:
I believe if the error is of type No rows found then you should continue, else for all other db type errors you should return it.